test/integration/cookbooks/docker-tests/recipes/default.rb

```diff
@@ -0,0 +1,25 @@
+package 'apparmor'
+package 'build-essential'
+
+include_recipe 'docker'
+ENV['CHEF_DRIVER'] = 'docker'
+
+bash 'build and install chef-provisioning-docker' do
+  cwd '/opt/chef-provisioning-docker'
+  code <<-EOS
+    rm chef-provisioning-docker-*.gem
+    /opt/chef/embedded/bin/gem build chef-provisioning-docker.gemspec
+  EOS
+  action :nothing
+end.run_action(:run)
+
+$:.unshift('/opt/chef-provisioning-docker/lib')
+require 'chef/provisioning/docker_driver/version'
+gem_dir = '/opt/chef-provisioning-docker'
+gem_path = File.join(gem_dir, "chef-provisioning-docker-#{Chef::Provisioning::DockerDriver::VERSION}.gem")
+gem_package 'chef-provisioning-docker' do
```

Comment:
Why not `chef_gem` ? Test-kitchen will install chef omnibus with embedded ruby on VM, so there should be no ruby installed in system. No sure how `gem_package` handle this, but `chef_gem` will install it using that embedded ruby.